python/python/lance/torch/data.py

```diff
@@ -111,7 +111,29 @@ def _to_tensor(
     return ret
 
 
-class TensorDataset(torch.utils.data.Dataset):
+logger = logging.getLogger(__name__)
+
+try:
+    # When available, subclass from the newer torchdata DataPipes
+    # instead of torch Datasets.
```

Comment:
IMO, here we can add some comments about how to use the torchdata datapipe. e.g. bump up to which version...